app/screens/gallery/footer/download_with_action/index.tsx

```
@@ -231,12 +232,18 @@ const DownloadWithAction = ({action, enableSecureFilePreview, item, onShareCallb
             const hasPermission = await hasWriteStoragePermission(intl);
 
             if (hasPermission) {
-                switch (item.type) {
+                // Re-check actual file type from mime_type/extension, since
+                // servers may send generic mime types like application/octet-stream
+                // for videos, causing them to be misclassified as 'file'.
+                const fileInfo = galleryItemToFileInfo(item);
+                const actualType = isVideo(fileInfo) ? 'video' : (isImage(fileInfo) ? 'image' : item.type);
```

ESLint

[error] Do not nest ternary expressions.

(no-nested-ternary)